### PR TITLE
Implementation of maximum pod logsize - Fixes issue 878

### LINKED ISSUE #878: Max size of pod logs
**Internal Tracking**: [#55381](https://app.shortcut.com/replicated/story/55381)

### Problem to solve
Setting a max size limit on the pod logs. This is to ensure that the support bundle is not too large, the existing limit of max age or max lines would not necessarily help with this as there could be lines which are extremely long and thus increase the bundle size.

### Impact
This could result in large support bundles which would be harder to obtain from customers. Also this was possible in both Ship and native schedulers so should exist to keep feature parity 

### Proposed Solution
Add `limits.maxSize` or `limits.maxBytes` field to the pod logs parameters

### PR DESCRIPTION
**Pull request by edgarlanting**
## Description, Motivation and Context

This PR introduces a new option to limit the size of a pod log when added to the bundle. This will make sure the support bundle will not grow to an unacceptable size and thus might contain information that is too old.

The maximum size of a pod log in a bundle is set to 5MB, and can be changed if we decide upon the need.

Changes include the sample support bundle, log collecting & corresponding tests. For more details please see the related codebase changes.

A Troubleshoot docs PR has been created at: https://github.com/replicatedhq/troubleshoot.sh/pull/457

Fixes #878 

## Checklist

- [X] New and existing tests pass locally with introduced changes.
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] The commit message(s) are informative and highlight any breaking changes
- [X] Any documentation required has been added/updated. 

## Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

Note: As this _only_ introduces a maximum size to the pod logs collected, it will still collect pod logs in a bundle.

### CODE REVIEW

File: pkg/collect/logs.go [hunk 1]
```diff
@@ -264,6 +264,18 @@ func setLogLimits(podLogOpts *corev1.PodLogOptions, limits *troubleshootv1beta2.
 	} else {
 		podLogOpts.TailLines = &limits.MaxLines
 	}
+
+	defaultMaxBytes := int64(5000000)
+	if limits.MaxBytes == 0 {
+		podLogOpts.LimitBytes = &defaultMaxBytes
+		return
+	}
```
Comment (banjoh) [inline pkg/collect/logs.go]: These lines are not necessary

File: pkg/collect/logs_test.go [hunk 2]
```diff
@@ -7,13 +7,13 @@ import (
...
 	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	testclient "k8s.io/client-go/kubernetes/fake"
 )
 
 func Test_setLogLimits(t *testing.T) {
+	maxBytes := int64(5000000)
```
Comment (banjoh) [inline pkg/collect/logs_test.go:16]: You can change this to a non default value (e.g 200) to test that case instead

File: pkg/collect/logs_test.go [hunk 3]
```diff
@@ -58,23 +69,27 @@ func Test_setLogLimits(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			req := require.New(t)
 
 			actual := corev1.PodLogOptions{}
 			setLogLimits(&actual, test.limits, convertMaxAgeToTime)
 
+			if test.expected.LimitBytes == &maxBytes {
```
Comment (banjoh) [inline pkg/collect/logs_test.go]: You can change this line to `if test.expected.LimitBytes != nil {`

File: pkg/collect/logs_test.go (hunk 2 above)
Comment (edgarlanting) [reply pkg/collect/logs_test.go:16]: I should have documented that in the PR comment, thanks for highlighting. I'll make a small change.

File: pkg/collect/logs_test.go (hunk 3 above)
Comment (edgarlanting) [reply pkg/collect/logs_test.go]: Done

**Review by divolgin — DISMISSED** [review]

File: pkg/collect/cluster_resources.go [hunk 4]
```diff
@@ -169,6 +169,7 @@ func (c *CollectClusterResources) Collect(progressChan chan<- interface{}) (Coll
 		for _, container := range allContainers {
 			limits := &troubleshootv1beta2.LogLimits{
 				MaxLines: 500,
+				MaxBytes: 5000000,
```
Comment (divolgin) [inline pkg/collect/cluster_resources.go:172]: One of these limits seems redundant now.  Which one of them will be honored?

File: pkg/collect/logs.go [hunk 5]
```diff
@@ -286,6 +286,12 @@ func setLogLimits(podLogOpts *corev1.PodLogOptions, limits *troubleshootv1beta2.
 	} else {
 		podLogOpts.TailLines = &limits.MaxLines
 	}
+
+	if limits.MaxBytes == 0 {
+		podLogOpts.LimitBytes = &limits.MaxBytes
+	} else {
+		podLogOpts.LimitBytes = &limits.MaxBytes
+	}
 }
```
Comment (xavpaice) [inline pkg/collect/logs.go:298]: what does this conditional do?

**Review by xavpaice — CHANGES_REQUESTED** [review]
one query

File: pkg/collect/logs.go (hunk 5 above)
Comment (edgarlanting) [reply pkg/collect/logs.go:298]: This was a temporary workaround during testing..This has been resolved.

**Review by xavpaice — APPROVED** [review]
LGTM.

Future improvement: add a test for a non-default maxBytes.

**Pull request merged**